sdcm/utils/benchmarks.py

```diff
@@ -0,0 +1,137 @@
+
+
+class ScyllaClusterBenchmarkManager(metaclass=Singleton):
+    def __init__(self):
+        """
+        This class is meant for gathering the benchmark results
+        of all the relevant nodes in the cluster and presenting
+        them in unified fashion.
+        Optionally, these can also be committed to remote
+        storage (S3? Argus?)
+        """
+        self._nodes: list["BaseNode"] = []
+        self._benchmark_runners: list[ScyllaNodeBenchmarkRunner] = []
+
+    def add_node(self, new_node: "BaseNode"):
+        if new_node.distro.is_debian_like:
+            self._benchmark_runners.append(ScyllaNodeBenchmarkRunner(new_node))
+        else:
+            LOGGER.info("Skipped installing benchmarking tools on a non-debian-like distro.")
+
+    def install_benchmark_tools(self):
+        for runner in self._benchmark_runners:
+            runner.install_benchmark_tools()
+
+    def run_benchmarks(self):
+        for runner in self._benchmark_runners:
+            runner.run_benchmarks()
+
+    @staticmethod
+    def _collect_benchmark_output():
+        ...
+
+
+class ScyllaNodeBenchmarkRunner:
+    def __init__(self, node: "BaseNode"):
+        """
+        This class is meant to install and run benchmarking
+        tools on given cluster nodes and collect the output.
+        """
+        self._node = node
+        self._remoter: RemoteCmdRunnerBase = node.remoter
+
+    def install_benchmark_tools(self) -> bool:
+        self._clone_sysbench_repo()
+        self._install_ubuntu_prerequisites()
+        self._build_and_install_sysbench()
+        return self._check_sysbench_installation()
+
+    def _install_ubuntu_prerequisites(self) -> bool:
+        package_list = ["make", "automake", "libtool", "pkg-config", "libaio-dev"]
+        try:
+            LOGGER.info("Installing Ubuntu prerequisites for the node benchmarks...")
+            for pkg in package_list:
+                self._node.install_package(pkg)
+            return True
+        except Exception:
+            LOGGER.warning("Failed to install Ubuntu prerequisites for the node benchmarking tools.")
+            raise
+
+    def _build_and_install_sysbench(self) -> bool:
+        build_and_install_script = """\
+            cd ./sysbench
+            ./autogen.sh
+            ./configure --without-mysql
+            make -j
+            make install
+        """
+        try:
+            result = self._remoter.sudo(shell_script_cmd(build_and_install_script))
+            LOGGER.debug("Build and install sysbench result: %s", result.stdout)
+            LOGGER.debug("Build and install sysbench errors: %s", result.stderr)
+            return True
+        except Exception:
+            LOGGER.warning("Failed to install sysbench from source.")
+            raise
+
+    def _clone_sysbench_repo(self):
+        try:
+            LOGGER.info("Cloning sysbench repo...")
+            clone_cmd = "git clone https://github.com/akopytov/sysbench.git"
```

Comment:
Define this var before try block and use it in all the log messages.

Reply:
Fixed with rebase